petab/parameters.py

```diff
@@ -117,13 +130,67 @@ def get_required_parameters_for_parameter_table(
 
     Returns:
         Set of parameter IDs which PEtab requires to be present in the
-        parameter table
+        parameter table. That is all {observable,noise}Parameters from the
+        measurement table as well as all parametric condition table overrides
+        that are not defined in the SBML model.
+    """
+
+    # use ordered dict as proxy for ordered set
+    parameter_ids = OrderedDict()
+
+    # Add parameters from measurement table, unless they are fixed parameters
+    def append_overrides(overrides):
+        for p in overrides:
+            if isinstance(p, str) and p not in condition_df.columns:
+                parameter_ids[p] = None
+
+    for _, row in measurement_df.iterrows():
+        # we trust that the number of overrides matches
+        append_overrides(measurements.split_parameter_replacement_list(
+            row.observableParameters))
+        append_overrides(measurements.split_parameter_replacement_list(
+            row.noiseParameters))
+
+    # Add condition table parametric overrides unless already defined in the
+    # SBML model
+    for p in conditions.get_parametric_overrides(condition_df):
+        if sbml_model.getParameter(p) is None:
+            parameter_ids[p] = None
+
+    return parameter_ids.keys()
+
+
+def get_valid_parameters_for_parameter_table(
+        sbml_model: libsbml.Model,
+        condition_df: pd.DataFrame,
+        measurement_df: pd.DataFrame) -> Set[str]:
     """
+    Get set of parameters which may be present inside the parameter table
+
+    Arguments:
+        sbml_model: PEtab SBML model
+        condition_df: PEtab condition table
+        measurement_df: PEtab measurement table
+
+    Returns:
+        Set of parameter IDs which PEtab allows to be present in the
+        parameter table.
+    """
+
+    # - grab all model parameters
```

Comment:
alternatively in docstring?

Reply:
That's rather implementation-specific, would keep it separate.